File: app/src/main/java/com/taskail/mixion/fragments/ProfileFragment.java
package com.taskail.mixion.fragments;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.taskail.mixion.R;
import com.taskail.mixion.interfaces.FragmentLifecycle;
import com.taskail.mixion.utils.SteemJUtil;

import io.reactivex.Completable;
import io.reactivex.android.plugins.RxAndroidPlugins;
import io.reactivex.disposables.CompositeDisposable;
import io.reactivex.schedulers.Schedulers;

/**
 *Created by ed on 9/30/17.
 */

public class ProfileFragment extends Fragment implements FragmentLifecycle{


    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_profile, container, false);
    }

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

    }


    @Override
    public void onPauseFragment() {

    }

    @Override
    public void onResumeFragment() {

    }
}

File: app/src/main/java/com/taskail/mixion/profile/ProfileFragment.kt
package com.taskail.mixion.profile

import android.os.Bundle
import android.support.v4.app.Fragment
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.taskail.mixion.BackPressedHandler
import com.taskail.mixion.R
import com.taskail.mixion.utils.*
import kotlinx.android.synthetic.main.fragment_profile.*

/**
 *Created by ed on 2/3/18.
 */

class ProfileFragment : Fragment(), BackPressedHandler{

    private val TAG = javaClass.simpleName
    companion object {
        @JvmStatic fun newInstance(): ProfileFragment{
            return ProfileFragment()
        }
    }

    interface Callback {
        fun onProfileClose()
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(R.layout.fragment_profile, container, false)

        return view
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        profileToolbar.setNavigationOnClickListener {
            onBackPressed()
        }
        profileContainer.fadeInAnimation()
    }

    override fun onBackPressed(): Boolean {
        closeProfileFragment()
        return true
    }

    private fun closeProfileFragment(){
        profileContainer.fadeOutAnimation(object : FadeOutCallBack {
            override fun onAnimationEnd() {
                val callback = callback()
                callback?.onProfileClose()
            }
        })
    }

    private fun callback(): Callback? {
        return getCallback(this, Callback::class.java)
    }
}